### PR TITLE
Fixed crash of sidebar group & scroll

### PR DESCRIPTION
**Pull request by edencakir**
@joshmadewell

### CODE REVIEW

File: src/screens/Profiles/UserLibrary/components/UserLibraryScreen/component.js [hunk 1]
```diff
@@ -320,4 +320,4 @@ export class UserLibraryScreenComponent extends React.Component {
       </View>
     );
   }
-}
+}
```
Comment (joshmadewell) [inline src/screens/Profiles/UserLibrary/components/UserLibraryScreen/component.js]: if you install an editor config plugin for your editor, new liens will be magically added for you :D

File: src/screens/Sidebar/SidebarScreen.js [hunk 2]
```diff
@@ -71,21 +72,25 @@ class SidebarScreen extends React.Component {
...
+          console.log(item);
+          return (
+            <SidebarListItem
+              onPress={() => {
+                navigation.navigate('');
+              }}
+              title={item.group.name}
+              imageURL={
+                (item.group.avatar &&
+                  (item.group.avatar.small ||
```
Comment (joshmadewell) [inline src/screens/Sidebar/SidebarScreen.js]: we should write a helper for this I think -- not anything you need to do but I'm just thinking in text :D

File: src/screens/Sidebar/SidebarScreen.js [hunk 3]
```diff
@@ -111,90 +116,88 @@ class SidebarScreen extends React.Component {
...
       <View style={{ backgroundColor: colors.listBackPurple }}>
-        <View>
-          <ProgressiveImage
-            hasOverlay
-            style={styles.headerCoverImage}
-            source={{ uri: 'https://fubukinofansub.files.wordpress.com/2011/12/cover-03-04.jpg' }}
+        <ProgressiveImage
+          hasOverlay
+          style={styles.headerCoverImage}
+          source={{ uri: 'https://fubukinofansub.files.wordpress.com/2011/12/cover-03-04.jpg' }}
```
Comment (joshmadewell) [inline src/screens/Sidebar/SidebarScreen.js]: is this temporary? if not then we should store this in the build so we don't have to make a network request.

File: src/screens/Sidebar/SidebarScreen.js [hunk 4]
```diff
@@ -111,90 +116,88 @@ class SidebarScreen extends React.Component {
...
+          contentContainerStyle={{ paddingBottom: 100 }}
+          sections={sectionListData}
+          keyExtractor={(item, index) => index}
+          ListHeaderComponent={() => <View height={10} />}
+          renderItem={() => <SidebarListItem />}
+          renderSectionHeader={({ section }) => (
+            <SidebarTitle title={section.title} style={{ marginTop: 0 }} />
+          )}
+          renderSectionFooter={({ section }) => {
+            // THIS FUNCTION IS NOT BEING INVOKED !?
```
Comment (joshmadewell) [inline src/screens/Sidebar/SidebarScreen.js]: make sure you're reading the right docs :P we're on `0.44` which doesn't support this yet :(

File: src/screens/Sidebar/SidebarScreen.js [hunk 5]
```diff
@@ -111,90 +116,88 @@ class SidebarScreen extends React.Component {
...
+          }}
+          removeClippedSubviews={false}
+          SectionSeparatorComponent={() => <View height={20} />}
+          ListFooterComponent={() => (
```
Comment (joshmadewell) [inline src/screens/Sidebar/SidebarScreen.js]: we should try not using inline lambdas as I think it's hurting performance: https://stackoverflow.com/questions/36677733/why-jsx-props-should-not-use-arrow-functions

I would assign a `renderFooter = () => { ... }` function on your class and use that instead.

don't change everything now unless you have the time but moving forward opt for no jsx lambdas

File: src/screens/Sidebar/SidebarScreen.js (hunk 3 above)
Comment (edencakir) [reply src/screens/Sidebar/SidebarScreen.js]: temporary!

**Pull request merged**